### ISSUE TITLE
Depending on child environmental variables being set

### ISSUE THREAD

**Issue by erik-ekberg-sg** [jesusdalvarado/rapids-v1#5]
This package depends on environmental variables being set for child packages, however, (1) this dependency is not clear and (2) this dependency may conflict with other parallel implementations of the same packages. I would recommend making clear what environmental variables need to be set and ideally writing environmental variables unique to this package so this package can be used in isolation. Some environmental variables are

1. `JESUS_MILL_CLOUDEVENTS_SOURCE`
2. `JESUS_MILL_SOPS_...`
3. `JESUS_MILL_LAMBDA_...`

**Comment by jesusdalvarado** [comment]
@erik-ekberg-sg I agree, exposing the env variables is a good change. I don't think we would have conflicts that would require to have other names for environmental variables, but it won't hurt, I'll also add those.

**Comment by jesusdalvarado** [comment]
@erik-ekberg-sg  the only thing I don't like about this setting independent env variables for the package is that it will force us to expose the same variables twice (see screenshot). The reason is because we are using this rapids-v1 package to test the services, but using a different approach when testing the api's (e.g. stripe-customer-context-v0/api-get-v0/.e2e.js). I guess we can keep this pattern until we unify a single way to test Apis and services using the same package rapids-v1.

<img width="1082" alt="Screen Shot 2021-11-22 at 17 13 38" src="https://user-images.githubusercontent.com/31553124/142896257-e4ce5417-89e0-4d86-a1d7-70c08541f088.png">

**Comment by erik-ekberg-sg** [comment]
@jesusdalvarado, as you have pointed out the `@1mill/lambda` package is that unifying interface to interact with AWS Lambda functions: whether that be API or "rapids" calls. The scope of this package in my mind is to add helpers that make this "rapids" implementation easier by abstracting the boilerplate around the `@1mill/lambda` package. This is accomplished by (1) pre-installing `@1mill/cloudevents` and `@1mill/lambda` and (2) abstracting setup `@1mill/lambda` configurations specific to the serverless-rapids (but not necessarily API calls).

Nevertheless, if somebody is using `@1mill/lambda`, `@1mill/sops`, `@aws-sdk/client-lambda`, `@aws-sdk/client-kms`, or any AWS based package even though they can all be configured using the same `AWS_ACCESS_KEY_ID` environmental variable they all represent different use cases that need to be configured independently. A good example would be the `@1mill/sops` package needed READ access to AWS KMS to decrypt `.sops` files while `@aws-sdk/client-ksm` needs to have ADMIN access to create and modify keys on-demand for whatever reason. This would imply different API keys with different permission levels for these different use cases; though, the user could choose to share if they wanted. In a similar fashion, people need to be able to configure this package in a predictable and documented manner so they can tailor it to their use case whatever that may be.
